File: Top 100 Codes/01-Getting Started/09-Prime number/Primenumber.py
# Prompt the user to input a number
num = int(input("Enter a number: "))

# Check if the number is prime
if num <= 1:
    print(f"{num} is not a Prime Number.")
else:
    is_prime = True

    # Check divisors from 2 to the square root of num
    for i in range(2, int(num**0.5) + 1):
        if num % i == 0:
            is_prime = False
            break

    # Output the result
    if is_prime:
        print(f"{num} is a Prime Number.")
    else:
        print(f"{num} is not a Prime Number.")

File: Top 100 Codes/01-Getting Started/09-Prime number/Primenumber.cpp
#include <iostream>
using namespace std;

int main() {
    // Variable to store the number
    int num;

    // Prompt the user to input a number
    cout << "Enter a number: ";
    cin >> num;

    // Check if the number is prime
    if (num <= 1) {
        cout << num << " is not a Prime Number." << endl;
    } else {
        bool isPrime = true;

        // Check divisors from 2 to sqrt(num)
        for (int i = 2; i * i <= num; i++) {
            if (num % i == 0) {
                isPrime = false;
                break;
            }
        }

        // Output the result
        if (isPrime) {
            cout << num << " is a Prime Number." << endl;
        } else {
            cout << num << " is not a Prime Number." << endl;
        }
    }

    return 0;
}
